### DISCUSSION TITLE
Set up PiVPN alongside Pi-hole

### DISCUSSION THREAD

**Discussion by Fat-ctrl** [DesktopECHO/Pi-hole-for-Android#86] [General]
Hi, can you install PiVPN (Wireguard) alongside Pi-hole as I have seen it is not possible (the original [post](https://github.com/pivpn/pivpn/issues/913)) and if not then can other solution (Tailscale, ZeroTier) be viable.

**Comment by DesktopECHO** [comment]
Hi, pretty much any solution that doesn't need a kernel module compiled should work.

Wireguard is a no-go for that reason, but the others you mention should be ok.

Never played with Wireguard-Go before, that may be another option as it runs in userspace.

**Reply by Fat-ctrl** [reply]
Thanks for the reply
